src/nvim/eval/funcs.c

```diff
@@ -627,6 +622,9 @@ static void f_bufnr(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   if (argvars[0].v_type == VAR_UNKNOWN) {
     buf = curbuf;
   } else {
+    // Don't use tv_get_buf_from_arg(); we continue if the buffer wasn't found
+    // and the second argument isn't zero, but we want to return early if the
+    // first argument isn't a string or number so only one error is shown.
```

Comment:
Is this because patch v8.2.1183 is not ported yet?

Reply:
I don't think so as Vim still has the behaviour. If you try `bufnr()` with a first argument that's not a string/number:
```vim
:echo bufnr([], 1)
```

In Vim you get:
```
E745: Using a List as a Number
E730: Using List as a String
-1
```

In Nvim you get:
```
E745: Expected a Number or a String, List found
-1
```